tests/snappi/pfc/files/helper.py

```diff
@@ -519,9 +531,19 @@ def __verify_results(rows,
                                   'Total TX dropped packets {} should be 0'.
                                   format(dropped_packets))
 
-        # Check if pause frames are counted in the PFC counters
-        for peer_port, prios in flow_port_config[1].items():
-            for prio in prios:
-                pfc_pause_rx_frames = get_pfc_frame_count(duthost, peer_port, prio)
+    # Check if the counters are incremented correctly
+    # If the class enable vector is set, then the PFC pause frames should be counted in the PFC counters
+    # If the class enable vector is not set, then the PFC pause frames should be dropped, and the
+    # egress queue count on the switch should be incremented as packets continue onto the next hop
+    for peer_port, prios in flow_port_config[1].items():
+        for prio in range(len(prios)):
+            pfc_pause_rx_frames = get_pfc_frame_count(duthost, peer_port, prios[prio], is_tx=False)
+            total_egress_packets, _ = get_egress_queue_count(duthost, peer_port, prios[prio])
+            if class_enable_vec == class_enable_vector.BIT_SET:
                 pytest_assert(pfc_pause_rx_frames > 0,
                               "PFC pause frames with zero source MAC are not counted in the PFC counters")
+            elif class_enable_vec == class_enable_vector.NO_BIT_SET:
+                pytest_assert(pfc_pause_rx_frames == 0,
+                              "PFC pause frames with no bit set in the class enable vector should be dropped")
+                pytest_assert(total_egress_packets == test_tx_frames[prio], "Queue counters should increment for \
```

Comment:
I would suggest to keep only generic comparisons in this func. any checks specific to a testcase can reside in the testcase file or a separate func here. eg. L544 doesn't make sense in the context of this testcase

Reply:
L544 doesn't apply to this test case - it was there before for an earlier test case